### ISSUE TITLE
Skaffold is not using the image repository project for Cloud Build

### ISSUE THREAD

**Issue by kschaab** [GoogleContainerTools/skaffold#7923]
Set gcloud default project to a specific project, deploy to cluster in a different gcloud project with the registry set to that project.  Using cloudbuild to build attempt build, it will utilize cloud build in the gcloud default project rather than the expected project.

### Expected behavior
Cloud build uses the container registry project for building

### Actual behavior
Cloud build uses the gcloud default project

### Information

- Skaffold version: v1.39.2
- Operating system: MacOS 12.6 (21G115)
- Installed via: skaffold.dev
- Contents of skaffold.yaml:
```yaml
# To learn more about the skaffold.yaml schema visit
# https://skaffold.dev/docs/references/yaml/
apiVersion: skaffold/v2beta28
kind: Config
build:
  tagPolicy:
    sha256: {}
  artifacts:
  - context: .
    image: go-hello-world
deploy:
  kubectl:
    manifests:
    - kubernetes-manifests/**
profiles:
- name: cloudbuild
  build:
    platforms: ["linux/amd64"]
    googleCloudBuild: {}
```

### Steps to reproduce the behavior

1. Set gcloud default project to a specific project
2. Deploy to cluster in a different gcloud project with the registry set to that project.  
3. Using cloud build to build attempt build, it will utilize cloud build in the gcloud default project rather than the expected project.

**Comment by kschaab** [comment]
It doesn't seem this is coming from the gcloud default project but maybe the ADC token.  Logs:

```
skaffold dev -v debug --port-forward --filename /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml --wait-for-deletions-max 2m0s --wait-for-connection --default-repo gcr.io/cloud-code-ide-testing --profile cloudbuild
DEBU[0000] skaffold API not starting as it's not requested  subtask=-1 task=DevLoop
INFO[0000] Skaffold &{Version:v1.39.2 ConfigVersion:skaffold/v2beta29 GitVersion: GitCommit:7ed1d722b53b55ac1c909f00d00a05b47d0964dd BuildDate:2022-08-16T14:30:06Z GoVersion:go1.17 Compiler:gc Platform:darwin/amd64 User:}  subtask=-1 task=DevLoop
INFO[0000] Loaded Skaffold defaults from "/Users/keithsc/.skaffold/config"  subtask=-1 task=DevLoop
DEBU[0000] config version out of date: upgrading to latest "skaffold/v2beta29"  subtask=-1 task=DevLoop
DEBU[0000] parsed 1 configs from configuration file /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml  subtask=-1 task=DevLoop
INFO[0000] applying profile: cloudbuild                  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field Build  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field artifacts  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field insecureRegistries  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field tagPolicy  subtask=-1 task=DevLoop
INFO[0000] no values found in profile for field TagPolicy, using original config values  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field platforms  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field BuildType  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field Test   subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field Deploy  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field DeployType  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field docker  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field helm   subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kpt    subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kubectl  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kustomize  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field statusCheck  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field statusCheckDeadlineSeconds  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kubeContext  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field logs   subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field prefix  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field jsonParse  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field fields  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field PortForward  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field ResourceSelector  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field allow  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field deny   subtask=-1 task=DevLoop
INFO[0000] map entry found when executing locate for &{go-hello-world . <nil> {0xc000a1a0b0 <nil> <nil> <nil> <nil> <nil> <nil>} [] {[] []} []} of type *latest.Artifact and pointer: 824635603136  subtask=-1 task=DevLoop
INFO[0000] map entry found when executing locate for &[kubernetes-manifests/**] of type *[]string and pointer: 824635603328  subtask=-1 task=DevLoop
DEBU[0000] Running command: [tput colors]                subtask=-1 task=DevLoop
DEBU[0000] Command output: [256
]                        subtask=-1 task=DevLoop
invalid skaffold config: source: /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml, in unnamed config[0] on line 14 column 5: source: /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml, in unnamed config[0] on line 14 column 5: Manifest file "kubernetes-manifests/**" referenced in skaffold config could not be found. Verify that file "kubernetes-manifests/**" referenced in config "/Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml" exists and the path and naming are correct.
source: /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml, in unnamed config[0] on line 14 column 5: source: /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml, in unnamed config[0] on line 14 column 5: Manifest file "kubernetes-manifests/**" referenced in skaffold config could not be found. Verify that file "kubernetes-manifests/**" referenced in config "/Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml" exists and the path and naming are correct.
❯ cd hello-world-1/hello-world-1/hello-world-3
❯ skaffold dev -v debug --port-forward --filename /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml --wait-for-deletions-max 2m0s --wait-for-connection --default-repo gcr.io/cloud-code-ide-testing --profile cloudbuild
DEBU[0000] skaffold API not starting as it's not requested  subtask=-1 task=DevLoop
INFO[0000] Skaffold &{Version:v1.39.2 ConfigVersion:skaffold/v2beta29 GitVersion: GitCommit:7ed1d722b53b55ac1c909f00d00a05b47d0964dd BuildDate:2022-08-16T14:30:06Z GoVersion:go1.17 Compiler:gc Platform:darwin/amd64 User:}  subtask=-1 task=DevLoop
INFO[0000] Loaded Skaffold defaults from "/Users/keithsc/.skaffold/config"  subtask=-1 task=DevLoop
DEBU[0000] config version out of date: upgrading to latest "skaffold/v2beta29"  subtask=-1 task=DevLoop
DEBU[0000] parsed 1 configs from configuration file /Users/keithsc/hello-world-1/hello-world-1/hello-world-3/skaffold.yaml  subtask=-1 task=DevLoop
INFO[0000] applying profile: cloudbuild                  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field Build  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field artifacts  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field insecureRegistries  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field tagPolicy  subtask=-1 task=DevLoop
INFO[0000] no values found in profile for field TagPolicy, using original config values  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field platforms  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field BuildType  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field Test   subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field Deploy  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field DeployType  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field docker  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field helm   subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kpt    subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kubectl  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kustomize  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field statusCheck  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field statusCheckDeadlineSeconds  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field kubeContext  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field logs   subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field prefix  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field jsonParse  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field fields  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field PortForward  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field ResourceSelector  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field allow  subtask=-1 task=DevLoop
DEBU[0000] overlaying profile on config for field deny   subtask=-1 task=DevLoop
INFO[0000] map entry found when executing locate for &{go-hello-world . <nil> {0xc000a60000 <nil> <nil> <nil> <nil> <nil> <nil>} [] {[] []} []} of type *latest.Artifact and pointer: 824643617728  subtask=-1 task=DevLoop
INFO[0000] Using kubectl context: gke_cloud-code-ide-testing_us-central1-a_mixed-arch-cluster  subtask=-1 task=DevLoop
DEBU[0000] Running command: [minikube version --output=json]  subtask=-1 task=DevLoop
DEBU[0000] config version out of date: upgrading to latest "skaffold/v2beta29"  subtask=-1 task=DevLoop
DEBU[0000] config version out of date: upgrading to latest "skaffold/v2beta29"  subtask=-1 task=DevLoop
DEBU[0000] Command output: [{"commit":"62e108c3dfdec8029a890ad6d8ef96b6461426dc","minikubeVersion":"v1.26.1"}
]  subtask=-1 task=DevLoop
DEBU[0000] Running command: [/Users/keithsc/homebrew/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin/minikube profile list -o json --user=skaffold]  subtask=-1 task=DevLoop
DEBU[0001] Command output: [{"invalid":[],"valid":[{"Name":"cloud-run-dev-internal","Status":"Stopped","Config":{"Name":"cloud-run-dev-internal","KeepContext":true,"EmbedCerts":false,"MinikubeISO":"","KicBaseImage":"gcr.io/k8s-minikube/kicbase:v0.0.20@sha256:0250dab3644403384bd54f566921c6b57138eecffbb861f9392feef9b2ec44f6","Memory":1987,"CPUs":2,"DiskSize":20000,"VMDriver":"","Driver":"docker","HyperkitVpnKitSock":"","HyperkitVSockPorts":[],"DockerEnv":null,"ContainerVolumeMounts":null,"InsecureRegistry":null,"RegistryMirror":null,"HostOnlyCIDR":"192.168.99.1/24","HypervVirtualSwitch":"","HypervUseExternalSwitch":false,"HypervExternalAdapter":"","KVMNetwork":"default","KVMQemuURI":"qemu:///system","KVMGPU":false,"KVMHidden":false,"KVMNUMACount":1,"APIServerPort":0,"DockerOpt":null,"DisableDriverMounts":false,"NFSShare":[],"NFSSharesRoot":"/nfsshares","UUID":"","NoVTXCheck":false,"DNSProxy":false,"HostDNSResolver":true,"HostOnlyNicType":"virtio","NatNicType":"virtio","SSHIPAddress":"","SSHUser":"root","SSHKey":"","SSHPort":22,"KubernetesConfig":{"KubernetesVersion":"v1.20.2","ClusterName":"cloud-run-dev-internal","Namespace":"default","APIServerName":"minikubeCA","APIServerNames":null,"APIServerIPs":null,"DNSDomain":"cluster.local","ContainerRuntime":"containerd","CRISocket":"","NetworkPlugin":"","FeatureGates":"","ServiceCIDR":"10.96.0.0/12","ImageRepository":"","LoadBalancerStartIP":"","LoadBalancerEndIP":"","CustomIngressCert":"","RegistryAliases":"","ExtraOptions":[{"Component":"kubelet","Key":"cni-conf-dir","Value":"/etc/cni/net.mk"}],"ShouldLoadCachedImages":true,"EnableDefaultCNI":false,"CNI":"","NodeIP":"","NodePort":8443,"NodeName":""},"Nodes":[{"Name":"","IP":"192.168.58.2","Port":8443,"KubernetesVersion":"v1.20.2","ContainerRuntime":"containerd","ControlPlane":true,"Worker":true}],"Addons":{"ambassador":false,"auto-pause":false,"csi-hostpath-driver":false,"dashboard":false,"default-storageclass":true,"efk":false,"freshpod":false,"gcp-auth":true,"gvisor":false,"helm-tiller":false,"ingress":false,"ingress-dns":false,"istio":false,"istio-provisioner":false,"kong":false,"kubevirt":false,"logviewer":false,"metallb":false,"metrics-server":false,"nvidia-driver-installer":false,"nvidia-gpu-device-plugin":false,"olm":false,"pod-security-policy":false,"portainer":false,"registry":false,"registry-aliases":false,"registry-creds":false,"storage-provisioner":true,"storage-provisioner-gluster":false,"volumesnapshots":false},"CustomAddonImages":null,"CustomAddonRegistries":null,"VerifyComponents":{"apiserver":true,"system_pods":true},"StartHostTimeout":360000000000,"ScheduledStop":{"InitiationTime":1646869619,"Duration":600000000000},"ExposedPorts":[],"ListenAddress":"","Network":"","Subnet":"","MultiNodeRequested":false,"ExtraDisks":0,"CertExpiration":94608000000000000,"Mount":false,"MountString":"","Mount9PVersion":"","MountGID":"","MountIP":"","MountMSize":0,"MountOptions":null,"MountPort":0,"MountType":"","MountUID":"","BinaryMirror":"","DisableOptimizations":false,"DisableMetrics":false,"CustomQemuFirmwarePath":""},"Active":false},{"Name":"minikube","Status":"Stopped","Config":{"Name":"minikube","KeepContext":false,"EmbedCerts":false,"MinikubeISO":"https://storage.googleapis.com/minikube/iso/minikube-v1.26.1-amd64.iso","KicBaseImage":"gcr.io/k8s-minikube/kicbase:v0.0.23@sha256:baf6d94b2050bcbecd98994e265cf965a4f4768978620ccf5227a6dcb75ade45","Memory":4000,"CPUs":2,"DiskSize":20000,"VMDriver":"","Driver":"hyperkit","HyperkitVpnKitSock":"","HyperkitVSockPorts":[],"DockerEnv":null,"ContainerVolumeMounts":null,"InsecureRegistry":null,"RegistryMirror":null,"HostOnlyCIDR":"192.168.99.1/24","HypervVirtualSwitch":"","HypervUseExternalSwitch":false,"HypervExternalAdapter":"","KVMNetwork":"default","KVMQemuURI":"qemu:///system","KVMGPU":false,"KVMHidden":false,"KVMNUMACount":1,"APIServerPort":0,"DockerOpt":null,"DisableDriverMounts":false,"NFSShare":[],"NFSSharesRoot":"/nfsshares","UUID":"","NoVTXCheck":false,"DNSProxy":false,"HostDNSResolver":true,"HostOnlyNicType":"virtio","NatNicType":"virtio","SSHIPAddress":"","SSHUser":"root","SSHKey":"","SSHPort":22,"KubernetesConfig":{"KubernetesVersion":"v1.20.7","ClusterName":"minikube","Namespace":"default","APIServerName":"minikubeCA","APIServerNames":null,"APIServerIPs":null,"DNSDomain":"cluster.local","ContainerRuntime":"docker","CRISocket":"","NetworkPlugin":"","FeatureGates":"","ServiceCIDR":"10.96.0.0/12","ImageRepository":"","LoadBalancerStartIP":"","LoadBalancerEndIP":"","CustomIngressCert":"","RegistryAliases":"","ExtraOptions":null,"ShouldLoadCachedImages":true,"EnableDefaultCNI":false,"CNI":"","NodeIP":"","NodePort":8443,"NodeName":""},"Nodes":[{"Name":"","IP":"192.168.64.2","Port":8443,"KubernetesVersion":"v1.20.7","ContainerRuntime":"docker","ControlPlane":true,"Worker":true}],"Addons":{"ambassador":false,"auto-pause":false,"csi-hostpath-driver":false,"dashboard":false,"default-storageclass":true,"efk":false,"freshpod":false,"gcp-auth":false,"gvisor":false,"helm-tiller":false,"ingress":false,"ingress-dns":false,"istio":false,"istio-provisioner":false,"kong":false,"kubevirt":false,"logviewer":false,"metallb":false,"metrics-server":false,"nvidia-driver-installer":false,"nvidia-gpu-device-plugin":false,"olm":false,"pod-security-policy":false,"portainer":false,"registry":false,"registry-aliases":false,"registry-creds":false,"storage-provisioner":true,"storage-provisioner-gluster":false,"volumesnapshots":false},"CustomAddonImages":{"GCPAuthWebhook":"k8s-minikube/gcp-auth-webhook:v0.0.8@sha256:26c7b2454f1c946d7c80839251d939606620f37c2f275be2796c1ffd96c438f6","KubeWebhookCertgen":"k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.0@sha256:f3b6b39a6062328c095337b4cadcefd1612348fdd5190b1dcbcb9b9e90bd8068"},"CustomAddonRegistries":{},"VerifyComponents":{"apiserver":true,"system_pods":true},"StartHostTimeout":360000000000,"ScheduledStop":null,"ExposedPorts":[],"ListenAddress":"","Network":"","Subnet":"","MultiNodeRequested":false,"ExtraDisks":0,"CertExpiration":94608000000000000,"Mount":false,"MountString":"","Mount9PVersion":"","MountGID":"","MountIP":"","MountMSize":0,"MountOptions":null,"MountPort":0,"MountType":"","MountUID":"","BinaryMirror":"","DisableOptimizations":false,"DisableMetrics":false,"CustomQemuFirmwarePath":""},"Active":true}]}]  subtask=-1 task=DevLoop
DEBU[0001] Running command: [/Users/keithsc/homebrew/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin/minikube profile list -o json --user=skaffold]  subtask=-1 task=DevLoop
DEBU[0001] Command output: [{"invalid":[],"valid":[{"Name":"cloud-run-dev-internal","Status":"Stopped","Config":{"Name":"cloud-run-dev-internal","KeepContext":true,"EmbedCerts":false,"MinikubeISO":"","KicBaseImage":"gcr.io/k8s-minikube/kicbase:v0.0.20@sha256:0250dab3644403384bd54f566921c6b57138eecffbb861f9392feef9b2ec44f6","Memory":1987,"CPUs":2,"DiskSize":20000,"VMDriver":"","Driver":"docker","HyperkitVpnKitSock":"","HyperkitVSockPorts":[],"DockerEnv":null,"ContainerVolumeMounts":null,"InsecureRegistry":null,"RegistryMirror":null,"HostOnlyCIDR":"192.168.99.1/24","HypervVirtualSwitch":"","HypervUseExternalSwitch":false,"HypervExternalAdapter":"","KVMNetwork":"default","KVMQemuURI":"qemu:///system","KVMGPU":false,"KVMHidden":false,"KVMNUMACount":1,"APIServerPort":0,"DockerOpt":null,"DisableDriverMounts":false,"NFSShare":[],"NFSSharesRoot":"/nfsshares","UUID":"","NoVTXCheck":false,"DNSProxy":false,"HostDNSResolver":true,"HostOnlyNicType":"virtio","NatNicType":"virtio","SSHIPAddress":"","SSHUser":"root","SSHKey":"","SSHPort":22,"KubernetesConfig":{"KubernetesVersion":"v1.20.2","ClusterName":"cloud-run-dev-internal","Namespace":"default","APIServerName":"minikubeCA","APIServerNames":null,"APIServerIPs":null,"DNSDomain":"cluster.local","ContainerRuntime":"containerd","CRISocket":"","NetworkPlugin":"","FeatureGates":"","ServiceCIDR":"10.96.0.0/12","ImageRepository":"","LoadBalancerStartIP":"","LoadBalancerEndIP":"","CustomIngressCert":"","RegistryAliases":"","ExtraOptions":[{"Component":"kubelet","Key":"cni-conf-dir","Value":"/etc/cni/net.mk"}],"ShouldLoadCachedImages":true,"EnableDefaultCNI":false,"CNI":"","NodeIP":"","NodePort":8443,"NodeName":""},"Nodes":[{"Name":"","IP":"192.168.58.2","Port":8443,"KubernetesVersion":"v1.20.2","ContainerRuntime":"containerd","ControlPlane":true,"Worker":true}],"Addons":{"ambassador":false,"auto-pause":false,"csi-hostpath-driver":false,"dashboard":false,"default-storageclass":true,"efk":false,"freshpod":false,"gcp-auth":true,"gvisor":false,"helm-tiller":false,"ingress":false,"ingress-dns":false,"istio":false,"istio-provisioner":false,"kong":false,"kubevirt":false,"logviewer":false,"metallb":false,"metrics-server":false,"nvidia-driver-installer":false,"nvidia-gpu-device-plugin":false,"olm":false,"pod-security-policy":false,"portainer":false,"registry":false,"registry-aliases":false,"registry-creds":false,"storage-provisioner":true,"storage-provisioner-gluster":false,"volumesnapshots":false},"CustomAddonImages":null,"CustomAddonRegistries":null,"VerifyComponents":{"apiserver":true,"system_pods":true},"StartHostTimeout":360000000000,"ScheduledStop":{"InitiationTime":1646869619,"Duration":600000000000},"ExposedPorts":[],"ListenAddress":"","Network":"","Subnet":"","MultiNodeRequested":false,"ExtraDisks":0,"CertExpiration":94608000000000000,"Mount":false,"MountString":"","Mount9PVersion":"","MountGID":"","MountIP":"","MountMSize":0,"MountOptions":null,"MountPort":0,"MountType":"","MountUID":"","BinaryMirror":"","DisableOptimizations":false,"DisableMetrics":false,"CustomQemuFirmwarePath":""},"Active":false},{"Name":"minikube","Status":"Stopped","Config":{"Name":"minikube","KeepContext":false,"EmbedCerts":false,"MinikubeISO":"https://storage.googleapis.com/minikube/iso/minikube-v1.26.1-amd64.iso","KicBaseImage":"gcr.io/k8s-minikube/kicbase:v0.0.23@sha256:baf6d94b2050bcbecd98994e265cf965a4f4768978620ccf5227a6dcb75ade45","Memory":4000,"CPUs":2,"DiskSize":20000,"VMDriver":"","Driver":"hyperkit","HyperkitVpnKitSock":"","HyperkitVSockPorts":[],"DockerEnv":null,"ContainerVolumeMounts":null,"InsecureRegistry":null,"RegistryMirror":null,"HostOnlyCIDR":"192.168.99.1/24","HypervVirtualSwitch":"","HypervUseExternalSwitch":false,"HypervExternalAdapter":"","KVMNetwork":"default","KVMQemuURI":"qemu:///system","KVMGPU":false,"KVMHidden":false,"KVMNUMACount":1,"APIServerPort":0,"DockerOpt":null,"DisableDriverMounts":false,"NFSShare":[],"NFSSharesRoot":"/nfsshares","UUID":"","NoVTXCheck":false,"DNSProxy":false,"HostDNSResolver":true,"HostOnlyNicType":"virtio","NatNicType":"virtio","SSHIPAddress":"","SSHUser":"root","SSHKey":"","SSHPort":22,"KubernetesConfig":{"KubernetesVersion":"v1.20.7","ClusterName":"minikube","Namespace":"default","APIServerName":"minikubeCA","APIServerNames":null,"APIServerIPs":null,"DNSDomain":"cluster.local","ContainerRuntime":"docker","CRISocket":"","NetworkPlugin":"","FeatureGates":"","ServiceCIDR":"10.96.0.0/12","ImageRepository":"","LoadBalancerStartIP":"","LoadBalancerEndIP":"","CustomIngressCert":"","RegistryAliases":"","ExtraOptions":null,"ShouldLoadCachedImages":true,"EnableDefaultCNI":false,"CNI":"","NodeIP":"","NodePort":8443,"NodeName":""},"Nodes":[{"Name":"","IP":"192.168.64.2","Port":8443,"KubernetesVersion":"v1.20.7","ContainerRuntime":"docker","ControlPlane":true,"Worker":true}],"Addons":{"ambassador":false,"auto-pause":false,"csi-hostpath-driver":false,"dashboard":false,"default-storageclass":true,"efk":false,"freshpod":false,"gcp-auth":false,"gvisor":false,"helm-tiller":false,"ingress":false,"ingress-dns":false,"istio":false,"istio-provisioner":false,"kong":false,"kubevirt":false,"logviewer":false,"metallb":false,"metrics-server":false,"nvidia-driver-installer":false,"nvidia-gpu-device-plugin":false,"olm":false,"pod-security-policy":false,"portainer":false,"registry":false,"registry-aliases":false,"registry-creds":false,"storage-provisioner":true,"storage-provisioner-gluster":false,"volumesnapshots":false},"CustomAddonImages":{"GCPAuthWebhook":"k8s-minikube/gcp-auth-webhook:v0.0.8@sha256:26c7b2454f1c946d7c80839251d939606620f37c2f275be2796c1ffd96c438f6","KubeWebhookCertgen":"k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.0@sha256:f3b6b39a6062328c095337b4cadcefd1612348fdd5190b1dcbcb9b9e90bd8068"},"CustomAddonRegistries":{},"VerifyComponents":{"apiserver":true,"system_pods":true},"StartHostTimeout":360000000000,"ScheduledStop":null,"ExposedPorts":[],"ListenAddress":"","Network":"","Subnet":"","MultiNodeRequested":false,"ExtraDisks":0,"CertExpiration":94608000000000000,"Mount":false,"MountString":"","Mount9PVersion":"","MountGID":"","MountIP":"","MountMSize":0,"MountOptions":null,"MountPort":0,"MountType":"","MountUID":"","BinaryMirror":"","DisableOptimizations":false,"DisableMetrics":false,"CustomQemuFirmwarePath":""},"Active":true}]}]  subtask=-1 task=DevLoop
DEBU[0001] setting Docker user agent to skaffold-v1.39.2  subtask=-1 task=DevLoop
DEBU[0001] CLI platforms provided: ""                    subtask=-1 task=DevLoop
DEBU[0001] getting client config for kubeContext: `gke_cloud-code-ide-testing_us-central1-a_mixed-arch-cluster`  subtask=-1 task=DevLoop
DEBU[0002] intersect matchers "linux/amd64,linux/arm64" and "darwin/amd64", result ""  subtask=-1 task=DevLoop
DEBU[0002] platforms detected from active kubernetes cluster nodes: "linux/amd64"  subtask=-1 task=DevLoop
DEBU[0002] intersect matchers "linux/amd64" and "linux/amd64", result "linux/amd64"  subtask=-1 task=DevLoop
DEBU[0002] platforms selected for artifact "go-hello-world": "linux/amd64"  subtask=-1 task=DevLoop
DEBU[0002] Using builder: google cloud                   subtask=-1 task=DevLoop
INFO[0002] build concurrency first set to 0 parsed from *gcb.Builder[0]  subtask=-1 task=DevLoop
INFO[0002] final build concurrency value is 0            subtask=-1 task=DevLoop
Listing files to watch...
 - go-hello-world
DEBU[0003] Found dependencies for dockerfile: [{go.mod /hello-world true 9 9} {go.sum /hello-world true 9 9} {. /hello-world true 13 13} {template /hello-world/template false 31 31}]  subtask=-1 task=DevLoop
INFO[0003] List generated in 1.535 second                subtask=-1 task=DevLoop
Generating tags...
 - go-hello-world -> gcr.io/cloud-code-ide-testing/go-hello-world:latest
INFO[0003] Tags generated in 116.073µs                   subtask=-1 task=Build
Checking cache...
DEBU[0003] Could not import artifact from Docker, building instead (import of missing images disabled)  subtask=-1 task=Build
 - go-hello-world: Not found. Building
INFO[0003] Cache check completed in 9.757263ms           subtask=-1 task=Build
Starting build...
Building [go-hello-world]...
Target platforms: [linux/amd64]
DEBU[0003] Running command: [gcloud auth print-access-token --format=json]  subtask=go-hello-world task=Build
DEBU[0004] Command output: [{
  "token": "redacted"
}
]  subtask=go-hello-world task=Build
INFO[0004] unable to retrieve token: oauth2: token expired and refresh token is not set  subtask=-1 task=DevLoop
INFO[0004] falling back to application default credentials  subtask=-1 task=DevLoop
DEBU[0004] project id set to cloud-code-ide-testing      subtask=go-hello-world task=Build
Pushing code to gs://cloud-code-ide-testing_cloudbuild/source/cloud-code-ide-testing-06957205-cddd-4748-946e-6fa52db6e42e.tar.gz
DEBU[0005] Running command: [tput colors]                subtask=-1 task=DevLoop
DEBU[0005] Command output: [256
]                        subtask=-1 task=DevLoop
build [go-hello-world] failed: error creating build: googleapi: Error 403: Cloud Build API has not been used in project 566054308921 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudbuild.googleapis.com/overview?project=566054308921 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.Help",
    "links": [
      {
        "description": "Google developers console API activation",
        "url": "https://console.developers.google.com/apis/api/cloudbuild.googleapis.com/overview?project=566054308921"
      }
    ]
  },
  {
    "@type": "type.googleapis.com/google.rpc.ErrorInfo",
    "domain": "googleapis.com",
    "metadata": {
      "consumer": "projects/566054308921",
      "service": "cloudbuild.googleapis.com"
    },
    "reason": "SERVICE_DISABLED"
  }
]
```

**Comment by gsquared94** [comment]
Skaffold has an issue https://github.com/GoogleContainerTools/skaffold/issues/6996 where it's always using Application Default Credentials. That uses the project defined in `quota_project_id` field defined in `application_default_credentials.json` for resource usage, and if that project doesn't have Cloud Build API enabled, then it'd fail. I think that you're running into this behavior, and it'll be fixed with fixing #6996

**Comment by gsquared94** [comment]
Closing as duplicate of #6996
